src/services/nativeMessaging.service.ts

```diff
@@ -41,16 +55,17 @@ export class NativeMessagingService {
     });
   }
 
-  private async messageHandler(msg: any) {
+  private async messageHandler(msg: OuterMessage) {
     const appId = msg.appId;
     const rawMessage = msg.message;
 
     // Request to setup secure encryption
-    if (rawMessage.command === "setupEncryption") {
+    if ("command" in rawMessage && rawMessage.command === "setupEncryption") {
       const remotePublicKey = Utils.fromB64ToArray(rawMessage.publicKey).buffer;
 
       // Valudate the UserId to ensure we are logged into the same account.
-      if (rawMessage.userId !== (await this.stateService.getUserId())) {
+      const userIds = Object.keys(await firstValueFrom(this.stateService.accounts));
```

Comment:
You just need to get the current set of accounts, right? If so, this could probably just be `this.stateService.accounts.getValue()` and cut out the need to import rxjs